src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5388,6 +5351,63 @@ static TypeWithState calculateResultType(TypeWithAnnotations targetTypeWithNulla
                 var resultType = TypeWithState.Create(targetType, resultState);
                 return resultType;
             }
+
+            static NullableFlowState getReferenceConversionResultState(TypeWithAnnotations targetType, TypeWithState operandType)
+            {
+                var state = operandType.State;
+                switch (state)
+                {
+                    case NullableFlowState.MaybeNull:
+                        if (operandType.Type?.IsTypeParameterDisallowingAnnotation() != true &&
+                            targetType.Type?.IsTypeParameterDisallowingAnnotation() == true)
+                        {
+                            return NullableFlowState.MaybeDefault;
+                        }
+                        break;
+                    case NullableFlowState.MaybeDefault:
+                        if (targetType.Type?.IsTypeParameterDisallowingAnnotation() == false)
+                        {
+                            return NullableFlowState.MaybeNull;
+                        }
+                        break;
+                }
+                return state;
+            }
+
+            // Converting to a less-derived type (object, interface, type parameter).
+            // If the operand is MaybeNull or MaybeNullEvenIfNotNullable, the result should be
```

Comment:
>MaybeNullEvenIfNotNullable [](start = 46, length = 26)

`MaybeDefault`?

Reply:
And elsewhere

---
In reply to: [347664509](https://github.com/dotnet/roslyn/pull/39778#discussion_r347664509) [](ancestors = 347664509)